Keypad: List recent transactions

dennisreimann opened the pull request:
Closes #5379. 

## List

I played with various forms of data and display and ended up with the list group instead of a table and omitting the invoice ID, because the space is rather limited. In its current form it degrades nicely on even smaller displays, wrapping the status to a second line and aligning it on the right (see screenshot).

The whole row is clickable and will bring the user to the invoice details — we can then incorparate the refund process from there or link to the receipt and come up with a new and better approach of handling the refund flow in a separate PR.

![trans](https://github.com/btcpayserver/btcpayserver/assets/886/4d159e2b-6499-4fa0-9b0d-7964c88e6037)

![grafik](https://github.com/btcpayserver/btcpayserver/assets/886/7e2e3a35-6e8e-454f-9ad3-d0e3a0cf8983)

## Empty state

![trans-none](https://github.com/btcpayserver/btcpayserver/assets/886/5d224c4a-9fb3-4799-a6f2-df20ca5b4e72)

dstrukt commented:
FUGGG yeah!

Will review this evening, was gonna say maybe the table header was necessary, but suppose because there are so few items and their contents are clear enough, we don't need it.

dstrukt reviewed (changes requested):
Overall, looks incredible! This is a great addition!

Still reviewing a bit, but few thoughts:

- [ ] Don't need the additional `Close` button on the footer of the modal imo

- [ ] Think we should push the `Refresh` button as an icon on the header OR as a row line item either above or below the transactions in the modal body (could stay text)

- [ ] I need to get you a proper transactions icon, was a placeholder, which works great, but it reads more like a menu icon (on me)

Minor tweak consideration:

- [ ] The transaction icon could be 40x40, slightly larger than the current 32x32 .. thoughts?

- [ ] Can we center align horizontally the icon based on if a store logo or POS title is present, e.g: the transaction icon sits at the top of both, 

<img width="591" alt="Screen Shot 2023-11-16 at 9 43 46 AM" src="https://github.com/btcpayserver/btcpayserver/assets/6250771/ae76d91e-abf9-48e9-bde6-57a81ae77750">
<img width="575" alt="Screen Shot 2023-11-16 at 9 44 23 AM" src="https://github.com/btcpayserver/btcpayserver/assets/6250771/b58c5fbe-4d0b-48e0-a5ec-50edef74db87">

dennisreimann commented:
Updated the list according to what we discussed, @dstrukt 

![Bildschirmfoto am 2023-11-20 um 11 07 59](https://github.com/btcpayserver/btcpayserver/assets/886/e6e7f4e7-40c9-4d5c-bbb6-4b305935b941)

## No border version

Also added a no border version for discussion.

![Bildschirmfoto am 2023-11-20 um 11 12 54](https://github.com/btcpayserver/btcpayserver/assets/886/8b67a7fb-fea6-4b01-9019-54d96cb14d85)

NicolasDorier commented:
Came for the sovereign open source payment processor. Stayed for the Keypad.

dstrukt commented:
Getting you a final review - upgrading to .NET8

dstrukt reviewed (approved):
tACK!

Might want to adjust some of our modal UI a bit in the future, and we can tackle the icons in one swoop once I wrap them up, but the two we currently do work well!

Have a slight preference for the borderless numbered buttons, but could go either way, thoughts?

pavlenex commented:
> Have a slight preference for the borderless numbered buttons

Same, but not a blocker for me either.

pavlenex reviewed (approved):
tACK

rawgir-7Xoqca-nezgej reviewed (commented):
![image](https://github.com/btcpayserver/btcpayserver/assets/66771104/c0b62877-189e-46b9-9b20-90f30f14704e)![image](https://github.com/btcpayserver/btcpayserver/assets/66771104/8059e5d9-7302-4e64-b2ce-b39dbd0bd159)

rawgir-7Xoqca-nezgej reviewed (commented):
![image](https://github.com/btcpayserver/btcpayserver/assets/66771104/c0b62877-189e-46b9-9b20-90f30f14704e)![image](https://github.com/btcpayserver/btcpayserver/assets/66771104/8059e5d9-7302-4e64-b2ce-b39dbd0bd159)
